astropy/units/quantity.py

```diff
@@ -459,6 +459,13 @@ def __new__(
         # To ensure array remains fast, we short-circuit it.
         value_unit = None
         if not isinstance(value, np.ndarray):
+            try:
+                from array_api_compat import is_array_api_obj
```

Comment:
I think this should use the same optional dependency check as those in https://github.com/astropy/astropy/blob/main/astropy/utils/compat/optional_deps.py